### PR TITLE
blog: local-first expo legend-state

### PR DESCRIPTION
**Pull request by thorwebdev**
Preview: https://zone-www-dot-com-git-thor-blog-local-first-expo-0e15ca-supabase.vercel.app/blog/local-first-expo-legend-state

### CODE REVIEW

File: apps/www/_blog/2024-09-23-local-first-expo-legend-state.mdx [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+
+  return <></>
+})
+```
+
+`observer` is the suggested way of consuming observables for the best performance and safety.
+
+It turns the entire component into an observing context - it automatically tracks observables for changes when `get()` is called, even from within hooks or helper functions.
+
+This mean, as long as realtime is enabled on the respective table, the component will automatically update when changes are made to the data!
```
Comment (dshukertjr) [inline apps/www/_blog/2024-09-23-local-first-expo-legend-state.mdx:246]:
```suggestion
This means, as long as realtime is enabled on the respective table, the component will automatically update when changes are made to the data!
```

File: apps/www/_blog/2024-09-23-local-first-expo-legend-state.mdx [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+
+Since Legend-State utilizes supabase-js under the hood, you can use [Supabase Auth](/docs/guides/auth) and [row level security](/docs/guides/database/postgres/row-level-security) to restrict access to the data.
+
+For a tutorial on how to add user management to your Expo React Native application, refer to [this guide](/docs/guides/getting-started/tutorials/with-expo-react-native).
+
+## Conclusion
+
+Legend-State and Supabase are a powerful combination for building local-first applications. Legend-State pairs nicely with supabase-js, Supabase AUth and Supabase Realtime, allowing you to tap into the full power of the Supabase Stack while building fast and delightful applications that work across web and mobile platforms.
+
+Want to learn more about LegendState? Refer to their [docs](https://legendapp.com/open-source/state/v3/) and make sure to follow Jay Meistrich on [Twitter](https://twitter.com/jaymeistrich)!
```
Comment (jmeistrich) [inline apps/www/_blog/2024-09-23-local-first-expo-legend-state.mdx:340]: Twitter account is jmeistrich

File: apps/www/_blog/2024-09-23-local-first-expo-legend-state.mdx [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+## How Legend-State makes it work
+
+A primary goal of Legend-State is to make automatic persisting and syncing both easy and very robust, as it's meant to be used to power all storage and sync of complex apps.
+
+Any changes made while offline are persisted between sessions to be retried whenever connected. To do this, the sync system subscribes to changes on an observable, then on change goes through a multi-step flow to ensure that changes are persisted and synced.
+
+1. Save the pending changes to local persistence.
+1. Save the changes to local persistence.
+1. Save the changes to remote persistence.
+1. On remote save, set any needed changes (like updatedAt) back into the observable and local persistence.
```
Comment (jmeistrich) [inline apps/www/_blog/2024-09-23-local-first-expo-legend-state.mdx:48]: Might want this to be updated_at to match Supabase casing

File: apps/www/_blog/2024-09-23-local-first-expo-legend-state.mdx [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+import { configureSynced } from '@legendapp/state/sync'
+import { ObservablePersistMMKV } from '@legendapp/state/persist-plugins/mmkv'
+
+const supabase = createClient(
+  process.env.EXPO_PUBLIC_SUPABASE_URL,
+  process.env.EXPO_PUBLIC_SUPABASE_ANON_KEY
+)
+
+// Create a configured sync function
+const customSynced = configureSynced(syncedSupabase, {
```
Comment (jmeistrich) [inline apps/www/_blog/2024-09-23-local-first-expo-legend-state.mdx:112]: You may want to add some explanation that `syncedSupabase` is the plugin for supabase and this configures it with defaults?

**Review by jmeistrich — COMMENTED** [review]
Looks good overall! Just a few comments.

File: apps/www/_blog/2024-09-23-local-first-expo-legend-state.mdx (hunk 2 above)
Comment (jmeistrich) [inline apps/www/_blog/2024-09-23-local-first-expo-legend-state.mdx:340]: LegendState should be Legend-State

**Review by dshukertjr — APPROVED** [review]
Other than prettier failing, LGTM!

**Pull request merged**